### PR TITLE
Add tutorial about basic usage of parameter-efficient finetuning

### PR DESCRIPTION
**Pull request by sxjscience**
*Description of changes:*

- [x] Add a basic tutorial about parameter-efficient finetuning in AutoMM and how to combine it with gradient checkpointing. In the example, we used the same dataset as in our multilingual example, and finetune the backbone with `ia3_bias`. The final performance is comparable to full-finetuning. Also, I tried to add the mT5-XL finetuning example in the tutorial but the CI crashed. We can thus add billion-scale model finetuning in a later tutorial.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 1]
```diff
@@ -1252,30 +1252,32 @@ def _top_k_average(
...
-                    self._model = self._load_state_dict(
-                        model=model,
-                        path=top_k_model_paths[0],
-                        prefix=prefix,
-                    )
-                    best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]
-                    for i in range(1, len(top_k_model_paths)):
-                        cand_avg_state_dict = average_checkpoints(
-                            checkpoint_paths=ingredients + [top_k_model_paths[i]],
+                    if len(ingredients) > 1:
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/predictor.py:1256]: Should be `if len(top_k_model_paths) > 1:` ?

File: docs/tutorials/multimodal/efficient_finetuning_basic.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+models that have two billion parameters (e.g., [google/mt5-xl](https://huggingface.co/google/mt5-xl)) with a single GPU in [AWS G4 instances](https://aws.amazon.com/ec2/instance-types/g4/). 
+To turn on gradient checkpointing, you can just set up `"model.hf_text.gradient_checkpointing"` to `True`.
+
+```{.python .input}
+from autogluon.multimodal import MultiModalPredictor
+
+predictor = MultiModalPredictor(label="label",
+                                path="multilingual_ia3_gradient_checkpoint")
+predictor.fit(train_en_df,
+              presets="multilingual",
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/efficient_finetuning_basic.md:107]: The model used here is still `deberta` instead of the `t5` mentioned in the above description.

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 1 above)
Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/predictor.py]: Good point. It was a bug.

File: docs/tutorials/multimodal/efficient_finetuning_basic.md (hunk 2 above)
Comment (sxjscience) [reply docs/tutorials/multimodal/efficient_finetuning_basic.md:109]: Revised the text. I tried mT5-XL but the CI failed to build the tutorial. May be we can try it again when we are adding few-shot learning example.

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 3]
```diff
@@ -1252,30 +1252,32 @@ def _top_k_average(
...
-                            # Add new ingredient
-                            ingredients.append(top_k_model_paths[i])
-                            best_score = cand_score
+                        best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]
+                        for i in range(1, len(top_k_model_paths)):
+                            cand_avg_state_dict = average_checkpoints(
+                                checkpoint_paths=ingredients + [top_k_model_paths[i]],
+                            )
+                            self._model = self._load_state_dict(
+                                model=self._model,
```
Comment (suzhoum) [inline multimodal/src/autogluon/multimodal/predictor.py:1272]: I might not understand this correctly, but should this be `model=model`

File: docs/tutorials/multimodal/index.rst [hunk 4]
```diff
@@ -56,6 +56,12 @@ In the following, we prepared a few tutorials to help you learn how to use `Auto
 
       How to use CLIP to extract embeddings for retrieval problem.
 
+   .. card::
+      :title: Parameter-Efficient Finetuning in AutoMM -- Basic Usage
+      :link: efficient_finetuning_basic.html
+
+      How to take advantage of larger foundation models with the help of parameter-efficient finetuning.
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/index.rst:63]: Is it better to highlight here (or in the title) that we use IA3 + BitFit? Because later we might have other tutorials that use more advanced parameter-efficient finetuning technique. But of course we could modify it at that time.

**Review by bryanyzhu — APPROVED** [review]
LGTM, very useful tutorial!

File: docs/tutorials/multimodal/customization.md [hunk 5]
```diff
@@ -179,18 +179,23 @@ predictor.fit(hyperparameters={"optimization.top_k_average_method": "uniform_sou
...
+A few parameter-efficient finetuning options. It supports finetuning only a small portion of parameters instead of one whole pretrained backbone.
 
 - `"bit_fit"`: bias parameters only. See [this paper](https://arxiv.org/pdf/2106.10199.pdf) for details.
 - `"norm_fit"`: normalization parameters + bias parameters. See [this paper](https://arxiv.org/pdf/2003.00152.pdf) for details.
 - `"lora"`: LoRA Adaptors. See [this paper](https://arxiv.org/pdf/2106.09685.pdf) for details.
 - `"lora_bias"`: LoRA Adaptors + bias parameters.
 - `"lora_norm"`: LoRA Adaptors + normalization parameters + bias parameters.
+- `"ia3"`: IA3 algorithm. See [this paper](https://arxiv.org/abs/2205.05638) for details.
+- `"ia3_bias"`: IA3 + bias parameters.
+- `"ia3_norm"`: IA3 + normalization parameters + bias parameters.
```
[the remark this answers is not in the document]
Comment (sxjscience) [reply docs/tutorials/multimodal/customization.md:191]: I just feel that "ia3_norm_bias" is too long so used "ia3_norm".

File: docs/tutorials/multimodal/index.rst (hunk 4 above)
Comment (sxjscience) [reply docs/tutorials/multimodal/index.rst:63]: Fixed and emphasized we are using IA^3 + BitFit.

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 3 above)
Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/predictor.py:1272]: Actually, the "Tab" is confusing the diff tool, this should be aligned with Line 1270 in the original code.

**Review by suzhoum — APPROVED** [review]
LGTM!

File: docs/tutorials/multimodal/efficient_finetuning_basic.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+[LoRA](https://arxiv.org/pdf/2106.09685.pdf), [Adapter](https://arxiv.org/abs/1902.00751), [MAM Adapter](https://arxiv.org/pdf/2110.04366.pdf), [IA^3](https://arxiv.org/abs/2205.05638)). 
+These techniques can effectively reduce the peak memory usage and model training time, while maintaining the performance.
+
+In this tutorial, we introduce how to apply parameter-efficient finetuning in MultiModalPredictor.
+We will reuse the same multilingual dataset as in :ref:`sec_automm_textprediction_multilingual` and train 
+models with the `"ia3_bias"` algorithm. `"ia3_bias"` is a parameter-efficient finetuning algorithm that combines IA^3 and BitFit.
+
+## Prepare Dataset
+
+The [Cross-Lingual Amazon Product Review Sentiment](https://webis.de/data/webis-cls-10.html) dataset contains Amazon product reviews in four languages.
```
Comment (FANGAreNotGnu) [inline docs/tutorials/multimodal/efficient_finetuning_basic.md:25]: More recent cross-lingual datasets (like XNLI) appear more frequently in recent papers, but are also harder to fit. We could also try this dataset for distillation tutorial if it takes less time to train.

**Review by FANGAreNotGnu — APPROVED** [review]
Have a minor comment. LGTM!

File: docs/tutorials/multimodal/efficient_finetuning_basic.md (hunk 6 above)
Comment (sxjscience) [reply docs/tutorials/multimodal/efficient_finetuning_basic.md:25]: Yes, we can try this dataset.
